crates/router/src/core/admin.rs

```diff
@@ -281,6 +281,25 @@ pub async fn create_merchant_account(
         .await
         .to_duplicate_response(errors::ApiErrorResponse::DuplicateMerchantAccount)?;
 
+    if let Some(api_key) = merchant_account.publishable_key.as_ref() {
+        let state = state.clone();
```

Comment:
should we clone only necessary data here?

Reply:
https://github.com/juspay/hyperswitch/pull/5168/files#r1661123714